test/new-e2e/pkg/e2e/suite.go

```diff
@@ -529,6 +531,35 @@ func (bs *BaseSuite[Env]) TearDownSuite() {
 	}
 }
 
+// GetRootOutputDir returns the root output directory for tests to store output files and artifacts.
+// The directory is created on the first call to this function and reused in future calls.
+//
+// See BaseSuite.GetTestOutputDir() for a function that returns a directory for the current test.
+//
+// See GetRootOutputDir() for details on the root directory creation.
+func (bs *BaseSuite[Env]) GetRootOutputDir() (string, error) {
+	if bs.testSessionOutputDir == "" {
+		outputRoot, err := GetRootOutputDir()
+		if err != nil {
+			return "", err
+		}
+		// Store the timestamped directory to be used by all tests in the suite
+		bs.testSessionOutputDir = outputRoot
+	}
+	return bs.testSessionOutputDir, nil
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
have you considered using `sync.Once` instead?

It would require adding a `sync.Once` variable to `BaseSuite` and then calling it to ensure we call this method only once per suite. It would allow to remove the if.

See an example in [this Go playground](https://go.dev/play/p/bTXZf0u1kDd)

Reply:
`testify.Suite` isn't generally concurrency safe, but it's easy to add.